### PR TITLE
Fixes #31415 - Expose DHCP's ping_free_ip option

### PR DESCRIPTION
**Pull request by ekohl**
Foreman 1.24 introduced this option and it should be exposed so users can change it.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -130,6 +130,10 @@
 #
 # $dhcp_subnets::               Subnets list to restrict DHCP management to
 #
+# $dhcp_ping_free_ip::          Perform ICMP and TCP ping when searching free IPs from the pool. This makes
```
Comment (evgeni) [inline manifests/init.pp:133]: What's a TCP ping?

**Review by evgeni — APPROVED** [review]
test failures *look* unrelated?

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [reply manifests/init.pp:133]: Copied from https://github.com/theforeman/smart-proxy/blob/7c818d0cf1606a7d16c8334ab4fd256370eac365/config/settings.d/dhcp.yml.example#L14-L17. https://github.com/theforeman/smart-proxy/blob/7c818d0cf1606a7d16c8334ab4fd256370eac365/modules/dhcp_common/pingable.rb#L5-L40 suggests it's basically a TCP connect on port 7 (echo service).

**Comment by ekohl** [comment]
> test failures _look_ unrelated?

Looks like this was missed in the migration to GHA:

https://github.com/theforeman/puppet-foreman_proxy/blob/5dbef5d9e2a084344aef690487dfd1ce092e6eb3/spec/acceptance/journald_spec.rb#L14-L18

Turns out it's broken with Docker CE but works with the Docker version in CentOS 7 itself. Since Github Actions also runs on Docker CE, it's broken there as well.

**Comment by ekohl** [comment]
https://github.com/theforeman/puppet-foreman_proxy/pull/636 should fix that.

File: manifests/init.pp (hunk 1 above)
Comment (evgeni) [reply manifests/init.pp:133]: cool, thanks!

**Comment by evgeni** [comment]
@ekohl #636 is merged, feel free to rebase and merge here :)

**Pull request merged**